plugins/modules/win_dns_zone.py

```diff
@@ -0,0 +1,186 @@
+author: Joe Zollo (@joezollo)
+requirements:
+  - This module requires Windows Server 2012R2 or Newer
+description:
+  - Manage Windows Server DNS Zones
+  - Adds, Removes and Modifies DNS Zones - Primary, Secondary, Forwarder & Stub
+  - Task should be delegated to a Windows DNS Server
+options:
+  name:
+    description:
+      - Fully qualified name of the DNS zone.
+    type: str
+    required: true
+  type:
+    description:
+      - Specifies the type of DNS zone.
+      - When l(type=secondary), the DNS server will immediately attempt to
+        perform a zone transfer from the servers in this list. If this initial
+        transfer fails, then the zone will be left in an unworkable state.
+        This module does not verify the initial transfer.
+    type: str
+    choices: [ primary, secondary, stub, forwarder ]
+  dynamic_update:
+    description:
+      - Specifies how a zone handles dynamic updates.
+      - Secure DNS updates are available only for Active Directory-integrated
+        zones.
+      - When not specified during new zone creation, Windows will default this
+        to l(none).
+    type: str
+    choices: [ secure, none, nonsecureandsecure ]
+  state:
+    description:
+      - Specifies the desired state of the DNS zone.
+      - When l(state=present) the module will attempt to create the specified
+        DNS zone if it does not already exist.
+      - When l(state=absent), the module will remove the specified DNS
+        zone and all subsequent DNS records.
+    type: str
+    default: present
+    choices: [ present, absent ]
+  forwarder_timeout:
+    description:
+      - Specifies a length of time, in seconds, that a DNS server waits for a
+        master server to resolve a query.
+      - Accepts integer values between 0 and 15.
+      - If the provided value is not valid, it will be omitted and a warning
+        will be issued.
+    type: int
+  replication:
+    description:
+      - Specifies the replication scope for the DNS zone.
+      - l(replication=forest) will replicate the DNS zone to all domain
+        controllers in the Active Directory forest.
+      - l(replication=domain) will replicate the DNS zone to all domain
+        controllers in the Active Directory domain.
+      - l(replication=none) disables Active Directory integration and
+        creates a local file with the name of the zone.
+      - This is the equivalent of selecting l(store the zone in Active
+        Directory) in the GUI.
+    type: str
+    choices: [ forest, domain, legacy, none ]
+  dns_servers:
+    description:
+      - Specifies an list of IP addresses of the master servers of the zone.
+      - DNS queries for a forwarded zone are sent to master servers.
+      - Required if l(type=secondary), l(type=forwarder) or l(type=stub),
+        otherwise ignored.
+      - At least one server is required.
+    elements: str
+    type: list
+'''
+
+EXAMPLES = r'''
+- name: Ensure primary zone is present
+  win_dns_zone:
```

Comment:
While the rest of the modules haven't been updated we seem to be standardising on using the fully qualified collection name here, so `community.windows.win_dns_zone`.

I'm planning on updating the rest of the modules sometime soon and it would be good to have this PR already do that.

Reply:
On it! Thank you!

Reply:
Resolved.